File: medpay-order/src/main/java/com/medpay/order/domain/Order.java
package com.medpay.order.domain;

import com.medpay.common.domain.TenantEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
public class Order extends TenantEntity {

    @Column(name = "order_no", nullable = false, unique = true, length = 50)
    private String orderNo;

    @Column(name = "patient_id", nullable = false)
    private UUID patientId;

    @Column(name = "doctor_id")
    private UUID doctorId;

    @Column(name = "prescription_id")
    private UUID prescriptionId;

    @Column(name = "order_type", nullable = false, length = 30)
    private String orderType;

    @Column(name = "order_source", length = 30)
    private String orderSource = "ONLINE";

    @Column(name = "total_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal totalAmount;

    @Column(name = "discount_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal discountAmount = BigDecimal.ZERO;

    @Column(name = "insurance_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal insuranceAmount = BigDecimal.ZERO;

    @Column(name = "self_pay_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal selfPayAmount = BigDecimal.ZERO;

    @Column(name = "status", nullable = false, length = 30)
    private String status = "CREATED";

    @Column(name = "paid_at")
    private LocalDateTime paidAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @Column(name = "cancelled_at")
    private LocalDateTime cancelledAt;

    @Column(name = "cancel_reason", columnDefinition = "TEXT")
    private String cancelReason;

    @Column(name = "appointment_date")
    private LocalDate appointmentDate;

    @Column(name = "appointment_time_start")
    private LocalTime appointmentTimeStart;

    @Column(name = "appointment_time_end")
    private LocalTime appointmentTimeEnd;

    @Column(name = "schedule_id")
    private UUID scheduleId;

    @Column(name = "delivery_type", length = 20)
    private String deliveryType;

    @Column(name = "delivery_address", columnDefinition = "TEXT")
    private String deliveryAddress;

    @Column(name = "delivery_status", length = 20)
    private String deliveryStatus;

    @Column(name = "remark", columnDefinition = "TEXT")
    private String remark;

    @Column(name = "expire_at")
    private LocalDateTime expireAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<OrderItem> items = new ArrayList<>();
}

File: medpay-web/src/types/order.ts
import type { OrderType, ItemType, OrderStatus, OrderSource } from './enums';

export interface OrderItemRequest {
  itemType: ItemType;
  referenceId: string;
  quantity: number;
}

export interface AppointmentInfo {
  scheduleId: string;
  appointmentDate: string;
}

export interface DeliveryInfo {
  type: string;
  address: string;
}

export interface OrderCreateRequest {
  orderType: OrderType;
  items: OrderItemRequest[];
  appointmentInfo?: AppointmentInfo;
  deliveryInfo?: DeliveryInfo;
  couponCode?: string;
  remark?: string;
}

export interface AppointmentOrderRequest {
  scheduleId: string;
  doctorId?: string;
}

export interface OrderItemResponse {
  id: string;
  itemType: ItemType;
  referenceId: string;
  name: string;
  specification: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
  insuranceAmount: number;
  selfPayAmount: number;
}

export interface OrderResponse {
  id: string;
  orderNo: string;
  patientId: string;
  doctorId: string | null;
  orderType: OrderType;
  orderSource: OrderSource;
  totalAmount: number;
  discountAmount: number;
  insuranceAmount: number;
  selfPayAmount: number;
  status: OrderStatus;
  items: OrderItemResponse[];
  paidAt: string | null;
  completedAt: string | null;
  cancelReason: string | null;
  expireAt: string | null;
  appointmentDate: string | null;
  appointmentTimeStart: string | null;
  appointmentTimeEnd: string | null;
  deliveryType: string | null;
  deliveryAddress: string | null;
  remark: string | null;
  createdAt: string;
}
